src/web/routes/application/flow-control/session-accessors/session-accessors.js

```diff
@@ -41,12 +41,22 @@ const getStateFromSession = getJourneySessionProp(STATE_KEY)
 
 const getJourneysFromSession = compose(toPairs, pathOr({}, JOURNEYS_PATH))
 
+const getAdditionalDataForStep = (req, step) => {
+  return pathOr({}, [ADDITIONAL_DATA_KEY, step.path], req.session)
+}
+
+const setAdditionalDataForStep = (req, step, stepData) => {
+  req.session[ADDITIONAL_DATA_KEY][step.path] = stepData
```

Comment:
Should this be set per journey? .. although I guess this might not work if the claim is global.

Reply:
To quote @gethyn1: 'no' :)